src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java

```diff
@@ -409,6 +412,26 @@ private boolean isCollectionSupportSortByString(@NonNull DocumentCollection coll
         return indices.stream().anyMatch(isIndexingSupportSortByString());
     }
 
+    private Predicate<Index> isIndexingSupportStartsWith() {
+        return index -> {
+            if (index instanceof RangeIndex) {
+                final RangeIndex rangeIndex = (RangeIndex) index;
+                return rangeIndex.getDataType() == DataType.String;
+            }
+
+            return false;
+        };
+    }
+
+    private boolean isCollectionSupportStartsWith(@NonNull DocumentCollection collection) {
+        final IndexingPolicy policy = collection.getIndexingPolicy();
+        final List<Index> indices = new ArrayList<>();
+
+        policy.getIncludedPaths().forEach(p -> indices.addAll(p.getIndexes()));
+
+        return indices.stream().anyMatch(isIndexingSupportStartsWith());
+    }
```

Comment:
As the method like `sortSupported` only check collections, Can we add new Class for validation part ?